pkg/backend/display/diff_test.go

```diff
@@ -137,3 +137,63 @@ func TestDiffEvents(t *testing.T) {
 		})
 	}
 }
+
+func TestHasPolicyViolations(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name                   string
+		events                 []engine.PolicyViolationEventPayload
+		hasMandatoryViolations bool
+	}{
+		{
+			name:                   "no policy violations",
+			events:                 []engine.PolicyViolationEventPayload{},
+			hasMandatoryViolations: false,
+		},
+		{
+			name: "only advisory violations",
+			events: []engine.PolicyViolationEventPayload{
+				{
+					EnforcementLevel: "advisory",
+				},
+				{
+					EnforcementLevel: "advisory",
+				},
```

Comment:
optional readability suggestion: the individual payloads can be one line each since they hold just one field each

```suggestion
			events: []engine.PolicyViolationEventPayload{
				{EnforcementLevel: "advisory"},
				{EnforcementLevel: "advisory"},
```